core/rawdb/accessors_l1_message.go

```diff
@@ -40,6 +40,34 @@ func ReadSyncedL1BlockNumber(db ethdb.Reader) *uint64 {
 	return &value
 }
 
+// WriteSyncedL1BlockNumber writes the highest synced L1 block number to the database.
+func WriteIncludedL1BlockNumber(db ethdb.KeyValueWriter, L1BlockNumber uint64) {
```

Comment:
The idea is that we should not store this in DB, worker should simply get the messages based on `parent.Hash()`. Check the last message included in the parent and start from that +1.